### PR TITLE
Port `ingest/transform-strain-names` to `augur curate transform-strain-name` [#1486]

### PR DESCRIPTION
**Pull request by genehack**
## Description of proposed changes

Port `ingest/transform-strain-names` to `augur curate transform-strain-name`, add tests. 

## Related issue(s)

#1486
https://github.com/nextstrain/ingest/issues/43

## Checklist

- [x] Checks pass
- [x] If making user-facing changes, add a message in [CHANGES.md](https://github.com/nextstrain/augur/blob/HEAD/CHANGES.md) summarizing the changes in this PR

### CODE REVIEW

**Comment by genehack** [comment]
@joverlee521 I wonder if `filter-strain-name` would be a better command name? This isn't really _transforming_ anything, it's dropping things that don't match.

**Comment by joverlee521** [comment]
> @joverlee521 I wonder if `filter-strain-name` would be a better command name? This isn't really _transforming_ anything, it's dropping things that don't match.

Yeah, I was debating on this when I wrote up the issue. I think eventually the command would be expanded to _transform_ strain names as described in https://github.com/nextstrain/ingest/issues/25.

**Comment by genehack** [comment]
> Yeah, I was debating on this when I wrote up the issue. I think eventually the command would be expanded to _transform_ strain names as described in #1515.

Yeah, okay, fair.

File: augur/curate/transform_strain_name.py [hunk 1]
```diff
@@ -1,50 +1,77 @@
...
-             "Strain names that do not match the pattern will be dropped.")
-    parser.add_argument("--backup-fields", nargs="*",
-        help="List of backup fields to use as strain name if the value in 'strain' " +
-             "does not match the strain regex pattern. " +
-             "If multiple fields are provided, will use the first field that has a non-empty string.")
+def transform_name(
+    record: dict,
+    index: int,
+    strain_name_pattern: re.Pattern,
+    backup_fields: List[str],
```
Comment (joverlee521) [inline augur/curate/transform_strain_name.py:18]: _nit_

`backup_fields` can potentially be `None` if no `--backup-fields` are provided right?

```suggestion
    backup_fields: Optional[List[str]],
```

OR

add a `default=[]` to `--backup-fields` argument.

**Review by joverlee521 — APPROVED** [review]

**Pull request merged**